okhttp/src/main/java/okhttp3/internal/cache/CacheInterceptor.kt

```diff
@@ -0,0 +1,267 @@
+        .build()
+
+    if (cache != null) {
+      if (HttpHeaders.hasBody(response) && CacheStrategy.isCacheable(response, networkRequest)) {
+        // Offer this request to the cache.
+        val cacheRequest = cache.put(response)
+        return cacheWritingResponse(cacheRequest, response)
+      }
+
+      if (HttpMethod.invalidatesCache(networkRequest.method())) {
+        try {
+          cache.remove(networkRequest)
+        } catch (ignored: IOException) {
+          // The cache cannot be written.
+        }
+      }
+    }
+
+    return response
+  }
+
+  /**
+   * Returns a new source that writes bytes to `cacheRequest` as they are read by the source
+   * consumer. This is careful to discard bytes left over when the stream is closed; otherwise we
+   * may never exhaust the source stream and therefore not complete the cached response.
+   */
+  @Throws(IOException::class)
+  private fun cacheWritingResponse(cacheRequest: CacheRequest?, response: Response): Response {
+    // Some apps return a null body; for compatibility we treat that like a null cache request.
+    if (cacheRequest == null) return response
+    val cacheBodyUnbuffered = cacheRequest.body()
+
+    val source = response.body()!!.source()
+    val cacheBody = cacheBodyUnbuffered.buffer()
+
+    val cacheWritingSource = object : Source {
+      var cacheRequestClosed: Boolean = false
+
+      @Throws(IOException::class)
+      override fun read(sink: Buffer, byteCount: Long): Long {
+        val bytesRead: Long
+        try {
+          bytesRead = source.read(sink, byteCount)
+        } catch (e: IOException) {
+          if (!cacheRequestClosed) {
+            cacheRequestClosed = true
+            cacheRequest.abort() // Failed to write a complete cache response.
+          }
+          throw e
+        }
+
+        if (bytesRead == -1L) {
+          if (!cacheRequestClosed) {
+            cacheRequestClosed = true
+            cacheBody.close() // The cache response is complete!
+          }
+          return -1
+        }
+
+        sink.copyTo(cacheBody.buffer, sink.size - bytesRead, bytesRead)
+        cacheBody.emitCompleteSegments()
+        return bytesRead
+      }
+
+      override fun timeout(): Timeout {
+        return source.timeout()
+      }
+
+      @Throws(IOException::class)
+      override fun close() {
+        if (!cacheRequestClosed && !discard(this, ExchangeCodec.DISCARD_STREAM_TIMEOUT_MILLIS,
+                MILLISECONDS)) {
+          cacheRequestClosed = true
+          cacheRequest.abort()
+        }
+        source.close()
+      }
+    }
+
+    val contentType = response.header("Content-Type")
+    val contentLength = response.body()?.contentLength() ?: -1
+    return response.newBuilder()
+        .body(RealResponseBody(contentType, contentLength, cacheWritingSource.buffer()))
+        .build()
+  }
+
+  companion object {
+
+    private fun stripBody(response: Response?): Response? {
+      if (response?.body() != null)
```

Comment:
braces on all if clauses and else clauses that wrap

Reply:
Fixed